### PR TITLE
StringBuffer -> StringBuilder and other modernizations

### PR DESCRIPTION
**Pull request by GlenKPeterson**
Thank you for adding a Maven project to https://github.com/djmdjm/jBCrypt

BCrypt needs to be efficient because it relies on "work" for security.  If someone else can make a faster version, they can break your encryption more cheaply than you can make it.  StringBuilder is a more efficient replacement for StringBuffer, so I wanted to make that change.

Using supported software is considered more secure than unsupported, so I moved us up to Java 8.

Added the Maven Enforcer plugin to make checking versions easier in the future.

Upgraded to the latest JUnit.

Added `CHANGELOG.md` and `.gitignore` files.

Basically I wanted to update things as securely as possible while changing as little as possible.  What do you think?

### CODE REVIEW

**Review by bobaikato — APPROVED** [review]

**Comment by v4ultboy** [comment]
I ended up using [Password4j](https://github.com/Password4j/password4j) which is frequently maintained.
